minecraft/forge/src/main/java/dev/kosmx/playerAnim/forge/ForgeClientEvent.java

```diff
@@ -5,31 +5,31 @@
 import dev.kosmx.playerAnim.minecraftApi.PlayerAnimationRegistry;
 import net.minecraft.server.packs.resources.ResourceManagerReloadListener;
 import net.neoforged.api.distmarker.Dist;
-import net.neoforged.fml.common.EventBusSubscriber;
+import net.neoforged.bus.api.IEventBus;
 import net.neoforged.neoforge.client.event.RegisterClientReloadListenersEvent;
-import net.neoforged.bus.api.SubscribeEvent;
 import net.neoforged.fml.ModList;
 import net.neoforged.fml.common.Mod;
-import net.neoforged.fml.event.lifecycle.FMLClientSetupEvent;
 import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@EventBusSubscriber(modid = "playeranimator", bus = EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
+@Mod(value = "playeranimator", dist = Dist.CLIENT)
 public class ForgeClientEvent {
+    public static final Logger LOGGER = LoggerFactory.getLogger("player-animator");
+
+    public ForgeClientEvent(IEventBus bus) {
+        bus.addListener(this::resourceLoadingListener);
 
-    @SubscribeEvent
-    public static void clientSetup(FMLClientSetupEvent event) {
         if (Helper.isBendEnabled() && ModList.get().isLoaded("skinlayers3d")) {
             try {
-                SkinLayersTransformer.init(ForgeModInterface.LOGGER);
+                SkinLayersTransformer.init(LOGGER);
             } catch(Error e) {
-                ForgeModInterface.LOGGER.error("Failed to initialize 3D skin layers compat: " + e.getMessage());
+                LOGGER.error("Failed to initialize 3D skin layers compat: " + e.getMessage());
             }
         }
     }
 
-    @SubscribeEvent
-    public static void resourceLoadingListener(@NotNull RegisterClientReloadListenersEvent event) {
-        event.registerReloadListener((ResourceManagerReloadListener) manager -> PlayerAnimationRegistry.resourceLoaderCallback(manager, ForgeModInterface.LOGGER));
+    public void resourceLoadingListener(@NotNull RegisterClientReloadListenersEvent event) {
```

Comment:
does this work if not static?
forge event bus is broken...

Reply:
There's a listener in the ctor, everything's fine.